Reworks Learnergroup selection component

stopfstedt opened the pull request:
![image](https://user-images.githubusercontent.com/1410427/188976119-c792ad32-e0b6-4e38-9e55-5f935052cf51.png)


- makes it more evident what is selected (and what is not) by using checkboxes in the picker.
- use tag lists for selected groups that contain the full parentage of each group, making selected parent groups explicit.
- fixes state handling/loading bug that was preventing cascading selects on first click in the picker
- adds ctrl+click/shift-click support to allow selection/deselection of groups with children without cascading effect.

fixes https://github.com/ilios/ilios/issues/3905
fixes https://github.com/ilios/ilios/issues/3896

dartajax reviewed (commented):
I think I found a bug. See below ...

These are sub groups of BRIDGES Class of 2026 >> Aardvark Island
I am going to select Aardvark Island >> Not Enough Ants >> Never Enough Ants first.

<img width="1194" alt="Screen Shot 2022-09-07 at 2 24 37 PM" src="https://user-images.githubusercontent.com/7493331/188984031-f71024a2-97d8-4128-9a1a-ca878dbf3458.png">

After performing the action shown above ...

![image](https://user-images.githubusercontent.com/7493331/188984601-ff801e9c-9420-4410-8505-defca2931030.png)

I stated in my screen shot that it was correct - It actually is not correct - I apologize for that ....

I selected `Aardvark Island >> Not Enough Ants >> Never Enough Antz`
It appears that I selected `Not Enough Ants >> Aardvark Island >> Never Enough Antz` - the subs and parents are not in the correct order.

Now I return to select as shown below ...

<img width="1028" alt="Screen Shot 2022-09-07 at 2 28 08 PM" src="https://user-images.githubusercontent.com/7493331/188984784-89963610-bbf4-4ec6-817a-201e18222566.png">

This should add the parent group so that now there are two groups selected but please review this screen shot for reference ...

![image](https://user-images.githubusercontent.com/7493331/188985331-c86a4da3-921e-4360-a724-b783e0430155.png)

The same group appears to have been added two times - This is pre-save though it should be pointed out. After Save ...

<img width="985" alt="Screen Shot 2022-09-07 at 2 32 05 PM" src="https://user-images.githubusercontent.com/7493331/188985725-4527eca0-5907-4227-8187-6defe154c9c1.png">

Only one of the two groups (not three) were added to the offering. I wanted to add the child and the parent to the offering. Try this by directly selecting the parent to implicitly also select the child.

![image](https://user-images.githubusercontent.com/7493331/188986504-eaa3a27a-4441-4c4d-8921-f1760beb8052.png)

After pressing "Save" ...

<img width="984" alt="Screen Shot 2022-09-07 at 2 36 30 PM" src="https://user-images.githubusercontent.com/7493331/188986708-6fa3d4e4-a564-4fef-9424-189c2b510382.png">


The child gets added but not the parent.

stopfstedt commented:
@dartajax i addressed the following bugs:

- corrected order of parent group titles  in selected groups tags (fixes sort order as well)
- de-duplicates selected groups
- prevent "knockout" effect that was yanking selected groups if they have selected descendants

please click-test again.

dartajax reviewed (commented):
All the things I reported in my previous comment have been fixed and look really nice ...

Here's another potential mini-caveat.

![image](https://user-images.githubusercontent.com/7493331/188997877-591d6109-b3ba-466b-b8f1-7c19cc327eec.png)

dartajax commented:
Now that we have check boxes, the default if a parent is selected is to have all children selected as well. That's fine but what if we just want the children and don't want to have to go back in a re-click each child. Each check box applies to that one group now so is that possible?

dartajax reviewed (approved):
I am good with this at least on my Shift using Mac.

jrjohnson commented on addon/components/learnergroup-tree.js:
```diff
@@ -58,10 +54,13 @@ export default class LearnergroupTree extends Component {
-      this.args.add(learnerGroup);
-    }
+  add(learnerGroup, ev) {
+    this.args.add(learnerGroup, !(ev.ctrlKey || ev.shiftKey));
```
Wow. That's 100x cleaner than I could have imagined. Nice!

jrjohnson commented on addon/models/learner-group.js:
```diff
@@ -336,7 +336,25 @@ export default class LearnerGroup extends Model {
+    return [...allParents, parent];
+  }
+
+  async getAllParentTitles() {
```
Why not `this.allParentTitles`?
